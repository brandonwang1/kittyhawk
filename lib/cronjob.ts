import { Construct } from 'constructs';
import { CronJob as CronJobApiObject } from '../imports/k8s';


export interface CronJobOptions {

    /**
     * CronJob name
     */
    readonly name: string;

    /**
     * CronJob schedule
     */
    readonly schedule: string;

    /**
     * CronJob image 
     * 
     */
    readonly image: string;

    /**
     * CronJob secrets
     * 
     * @default undefined
     * 
     */
    readonly secret?: string;

    /**
    * CronJob commands
    * 
    * @default []
    * 
    */
    readonly cmd: string[];

    /**
    * CronJob restart policy
    * 
    * @default "Never"
    * 
    */
    readonly restartPolicy?: "Always" | "OnFailure" | "Never";

    /**
    * CronJob successLimit
    * 
    * @default 1
    * 
    */
    readonly successLimit?: number;

    /**
    * CronJob failureLimit
    * 
    * @default 1
    * 
    */
    readonly failureLimit?: number;

}


export class CronJob extends Construct {
    constructor(scope: Construct, id: string, options: CronJobOptions) {
        super(scope, id);

        const name = options.name;
        const schedule = options.schedule;
        // const restartPolicy = options.restartPolicy || "Never";
        const failureLimit = options.failureLimit ?? 1;
        const successLimit = options.successLimit ?? 1;

        new CronJobApiObject(this, `cronjob-${name}`, {
            spec: {
                schedule: schedule,
                jobTemplate: {},
                failedJobsHistoryLimit: failureLimit,
                successfulJobsHistoryLimit: successLimit,
            }
        });
    }
}