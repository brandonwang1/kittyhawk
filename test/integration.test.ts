import { Construct } from 'constructs';
import { CronJob } from '../lib/cronjob';
import { Application, DjangoApplication, ReactApplication } from '../lib/application'
import { chartTest } from './utils'
import cronTime from 'cron-time-generator';

/** INTEGRATION TEST CONFIGS */

export function buildWebsiteChart(scope: Construct) {

  /** Penn Labs Website **/
  new Application(scope, 'serve', {
    image: 'pennlabs/website',
    ingress: [{ host: 'pennlabs.org', paths: ['/'] }],
  })
}

export function buildBasicsChart(scope: Construct) {

  /** Penn Basics **/
  new Application(scope, 'react', {
    image: 'pennlabs/penn-basics',
    secret: 'penn-basics',
    extraEnv: [{ name: 'PORT', value: '80' }],
    ingress: [{ host: 'pennbasics.com', paths: ['/'] }],
  })
}

export function buildOHQChart(scope: Construct) {

  /** OHQ (Part of it) **/
  new DjangoApplication(scope, 'django-asgi', {
    image: 'pennlabs/office-hours-queue-backend',
    secret: 'office-hours-queue',
    cmd: ['/usr/local/bin/asgi-run'],
    replicas: 2,
    domain: 'ohq.io',
    ingressPaths: ['/api/ws'],
    djangoSettingsModule: 'officehoursqueue.settings.production',
    extraEnv: [
      { name: 'REDIS_URL', value: 'redis://office-hours-queue-redis:6379' }],
  })

  new CronJob(scope, 'calculate-waits', {
    schedule: cronTime.every(5).minutes(),
    image: 'pennlabs/office-hours-queue-backend',
    secret: 'office-hours-queue',
    cmd: ['python', 'manage.py', 'calculatewaittimes'],
  });

}

export function buildCoursesChart(scope: Construct) {

  /** Penn Courses (Part of it) **/
  new Application(scope, 'backend', {
    image: 'pennlabs/penn-courses-backend',
    secret: 'penn-courses',
    cmd: ['celery', 'worker', '-A', 'PennCourses', '-Q', 'alerts,celery', '-linfo'],
    replicas: 3,
    extraEnv: [{ name: 'PORT', value: '80' },
      { name: 'DJANGO_SETTINGS_MODULE', value: 'PennCourses.settings.production' }],
    ingress: [
      { host: 'penncourseplan.com', paths: ['/api', '/admin', '/accounts', '/assets'] },
      { host: 'penncoursealert.com', paths: ['/api', '/admin', '/accounts', '/assets', '/webhook'] },
      { host: 'review.penncourses.org', paths: ['/api', '/admin', '/accounts', '/assets'] },
    ],
  })
}

export function buildPlatformChart(scope: Construct) {

  /** Platform **/
  new DjangoApplication(scope, 'platform', {
    image: 'pennlabs/platform',
    secret: 'platform',
    port: 443,
    domain: 'platform.pennlabs.org',
    djangoSettingsModule: 'Platform.settings.production',
    ingressPaths: ['/'],
    secretMounts: [{ name: 'platform', subPath: 'SHIBBOLETH_CERT', mountPath: '/etc/shibboleth/sp-cert.pem' },
      { name: 'platform', subPath: 'SHIBBOLETH_KEY', mountPath: '/etc/shibboleth/sp-key.pem' }],
  })
}

export function buildClubsChart(scope: Construct) {

  /** Penn Clubs **/
  new DjangoApplication(scope, 'django-asgi', {
    image: 'pennlabs/penn-clubs-backend',
    secret: 'penn-clubs',
    cmd: ['/usr/local/bin/asgi-run'],
    replicas: 2,
    domain: 'pennclubs.com',
    ingressPaths: ['/api/ws'],
    djangoSettingsModule: 'pennclubs.settings.production',
    extraEnv: [
      { name: 'REDIS_HOST', value: 'penn-clubs-redis' }],
  })

  new ReactApplication(scope, 'react', {
    image: 'pennlabs/penn-clubs-frontend',
    replicas: 2,
    domain: 'pennclubs.com',
    ingressPaths: ['/'],
    extraEnv: [{ name: 'PORT', value: '80' }],
  })

}


test('Penn Labs Website', () => chartTest(buildWebsiteChart));
test('Penn Basics', () => chartTest(buildBasicsChart));
test('OHQ', () => chartTest(buildOHQChart));
test('Penn Courses', () => chartTest(buildCoursesChart));
test('Platform API', () => chartTest(buildPlatformChart));
test('Penn Clubs', () => chartTest(buildClubsChart));

