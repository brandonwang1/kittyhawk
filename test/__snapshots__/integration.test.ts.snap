// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OHQ 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-django-asgi",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "name": "RELEASE_NAME-django-asgi",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-django-asgi",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-django-asgi",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 2,
      "selector": Object {
        "matchLabels": Object {
          "name": "RELEASE_NAME-django-asgi",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 3,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "name": "RELEASE_NAME-django-asgi",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "/usr/local/bin/asgi-run",
              ],
              "env": Array [
                Object {
                  "name": "DJANGO_SETTINGS_MODULE",
                  "value": "officehoursqueue.settings.production",
                },
                Object {
                  "name": "REDIS_URL",
                  "value": "redis://office-hours-queue-redis:6379",
                },
                Object {
                  "name": "DOMAIN",
                  "value": "ohq.io",
                },
              ],
              "envFrom": Array [
                Object {
                  "secretRef": Object {
                    "name": "office-hours-queue",
                  },
                },
              ],
              "image": "pennlabs/office-hours-queue-backend:TAG_FROM_CI",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-django-asgi",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "ohq.io",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-django-asgi",
                  "servicePort": 80,
                },
                "path": "/api/ws",
              },
            ],
          },
        },
      ],
      "tls": Array [
        Object {
          "hosts": Array [
            "ohq.io",
          ],
          "secretName": "ohq-io-tls",
        },
      ],
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "ohq-io",
    },
    "spec": Object {
      "dnsNames": Array [
        "ohq.io",
        "*.ohq.io",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "ohq-io-tls",
    },
  },
  Object {
    "apiVersion": "batch/v1beta1",
    "kind": "CronJob",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-calculate-waits",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-calculate-waits",
      "namespace": "default",
    },
    "spec": Object {
      "failedJobsHistoryLimit": 1,
      "jobTemplate": Object {
        "spec": Object {
          "template": Object {
            "spec": Object {
              "containers": Array [
                Object {
                  "command": Array [
                    "python",
                    "manage.py",
                    "calculatewaittimes",
                  ],
                  "envFrom": Array [
                    Object {
                      "secretRef": Object {
                        "name": "office-hours-queue",
                      },
                    },
                  ],
                  "image": "pennlabs/office-hours-queue-backend:TAG_FROM_CI",
                  "imagePullPolicy": "IfNotPresent",
                  "name": "worker",
                  "ports": Array [
                    Object {
                      "containerPort": 80,
                    },
                  ],
                },
              ],
              "restartPolicy": "Never",
            },
          },
        },
      },
      "schedule": "*/5 * * * *",
      "successfulJobsHistoryLimit": 1,
    },
  },
]
`;

exports[`Penn Basics 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-react",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "name": "RELEASE_NAME-react",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-react",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-react",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "name": "RELEASE_NAME-react",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 3,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "name": "RELEASE_NAME-react",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "PORT",
                  "value": "80",
                },
              ],
              "envFrom": Array [
                Object {
                  "secretRef": Object {
                    "name": "penn-basics",
                  },
                },
              ],
              "image": "pennlabs/penn-basics:TAG_FROM_CI",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-react",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "pennbasics.com",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-react",
                  "servicePort": 80,
                },
                "path": "/",
              },
            ],
          },
        },
      ],
      "tls": Array [
        Object {
          "hosts": Array [
            "pennbasics.com",
          ],
          "secretName": "pennbasics-com-tls",
        },
      ],
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "pennbasics-com",
    },
    "spec": Object {
      "dnsNames": Array [
        "pennbasics.com",
        "*.pennbasics.com",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "pennbasics-com-tls",
    },
  },
]
`;

exports[`Penn Clubs 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-django-asgi",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "name": "RELEASE_NAME-django-asgi",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-django-asgi",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-django-asgi",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 2,
      "selector": Object {
        "matchLabels": Object {
          "name": "RELEASE_NAME-django-asgi",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 3,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "name": "RELEASE_NAME-django-asgi",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "/usr/local/bin/asgi-run",
              ],
              "env": Array [
                Object {
                  "name": "DJANGO_SETTINGS_MODULE",
                  "value": "pennclubs.settings.production",
                },
                Object {
                  "name": "REDIS_HOST",
                  "value": "penn-clubs-redis",
                },
                Object {
                  "name": "DOMAIN",
                  "value": "pennclubs.com",
                },
              ],
              "envFrom": Array [
                Object {
                  "secretRef": Object {
                    "name": "penn-clubs",
                  },
                },
              ],
              "image": "pennlabs/penn-clubs-backend:TAG_FROM_CI",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-django-asgi",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "pennclubs.com",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-django-asgi",
                  "servicePort": 80,
                },
                "path": "/api/ws",
              },
            ],
          },
        },
      ],
      "tls": Array [
        Object {
          "hosts": Array [
            "pennclubs.com",
          ],
          "secretName": "pennclubs-com-tls",
        },
      ],
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "pennclubs-com",
    },
    "spec": Object {
      "dnsNames": Array [
        "pennclubs.com",
        "*.pennclubs.com",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "pennclubs-com-tls",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-react",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "name": "RELEASE_NAME-react",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-react",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-react",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 2,
      "selector": Object {
        "matchLabels": Object {
          "name": "RELEASE_NAME-react",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 3,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "name": "RELEASE_NAME-react",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "PORT",
                  "value": "80",
                },
                Object {
                  "name": "DOMAIN",
                  "value": "pennclubs.com",
                },
              ],
              "image": "pennlabs/penn-clubs-frontend:TAG_FROM_CI",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-react",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "pennclubs.com",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-react",
                  "servicePort": 80,
                },
                "path": "/",
              },
            ],
          },
        },
      ],
      "tls": Array [
        Object {
          "hosts": Array [
            "pennclubs.com",
          ],
          "secretName": "pennclubs-com-tls",
        },
      ],
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "pennclubs-com",
    },
    "spec": Object {
      "dnsNames": Array [
        "pennclubs.com",
        "*.pennclubs.com",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "pennclubs-com-tls",
    },
  },
]
`;

exports[`Penn Courses 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-backend",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "name": "RELEASE_NAME-backend",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-backend",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-backend",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 3,
      "selector": Object {
        "matchLabels": Object {
          "name": "RELEASE_NAME-backend",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 3,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "name": "RELEASE_NAME-backend",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "celery",
                "worker",
                "-A",
                "PennCourses",
                "-Q",
                "alerts,celery",
                "-linfo",
              ],
              "env": Array [
                Object {
                  "name": "PORT",
                  "value": "80",
                },
                Object {
                  "name": "DJANGO_SETTINGS_MODULE",
                  "value": "PennCourses.settings.production",
                },
              ],
              "envFrom": Array [
                Object {
                  "secretRef": Object {
                    "name": "penn-courses",
                  },
                },
              ],
              "image": "pennlabs/penn-courses-backend:TAG_FROM_CI",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-backend",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "penncourseplan.com",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/api",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/admin",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/accounts",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/assets",
              },
            ],
          },
        },
        Object {
          "host": "penncoursealert.com",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/api",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/admin",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/accounts",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/assets",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/webhook",
              },
            ],
          },
        },
        Object {
          "host": "review.penncourses.org",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/api",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/admin",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/accounts",
              },
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-backend",
                  "servicePort": 80,
                },
                "path": "/assets",
              },
            ],
          },
        },
      ],
      "tls": Array [
        Object {
          "hosts": Array [
            "penncourseplan.com",
          ],
          "secretName": "penncourseplan-com-tls",
        },
        Object {
          "hosts": Array [
            "penncoursealert.com",
          ],
          "secretName": "penncoursealert-com-tls",
        },
        Object {
          "hosts": Array [
            "review.penncourses.org",
          ],
          "secretName": "penncourses-org-tls",
        },
      ],
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "penncourseplan-com",
    },
    "spec": Object {
      "dnsNames": Array [
        "penncourseplan.com",
        "*.penncourseplan.com",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "penncourseplan-com-tls",
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "penncoursealert-com",
    },
    "spec": Object {
      "dnsNames": Array [
        "penncoursealert.com",
        "*.penncoursealert.com",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "penncoursealert-com-tls",
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "penncourses-org",
    },
    "spec": Object {
      "dnsNames": Array [
        "penncourses.org",
        "*.penncourses.org",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "penncourses-org-tls",
    },
  },
]
`;

exports[`Penn Labs Website 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-serve",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "name": "RELEASE_NAME-serve",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-serve",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-serve",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "name": "RELEASE_NAME-serve",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 3,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "name": "RELEASE_NAME-serve",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "pennlabs/website:TAG_FROM_CI",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-serve",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "pennlabs.org",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-serve",
                  "servicePort": 80,
                },
                "path": "/",
              },
            ],
          },
        },
      ],
      "tls": Array [
        Object {
          "hosts": Array [
            "pennlabs.org",
          ],
          "secretName": "pennlabs-org-tls",
        },
      ],
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "pennlabs-org",
    },
    "spec": Object {
      "dnsNames": Array [
        "pennlabs.org",
        "*.pennlabs.org",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "pennlabs-org-tls",
    },
  },
]
`;

exports[`Platform API 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-platform",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 443,
          "targetPort": 443,
        },
      ],
      "selector": Object {
        "name": "RELEASE_NAME-platform",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "name": "RELEASE_NAME-platform",
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-platform",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "name": "RELEASE_NAME-platform",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 3,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "name": "RELEASE_NAME-platform",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "DJANGO_SETTINGS_MODULE",
                  "value": "Platform.settings.production",
                },
                Object {
                  "name": "DOMAIN",
                  "value": "platform.pennlabs.org",
                },
              ],
              "envFrom": Array [
                Object {
                  "secretRef": Object {
                    "name": "platform",
                  },
                },
              ],
              "image": "pennlabs/platform:TAG_FROM_CI",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": Array [
                Object {
                  "containerPort": 443,
                },
              ],
              "volumeMounts": Array [
                Object {
                  "mountPath": "/etc/shibboleth/sp-cert.pem",
                  "name": "platform",
                  "subPath": "SHIBBOLETH_CERT",
                },
                Object {
                  "mountPath": "/etc/shibboleth/sp-key.pem",
                  "name": "platform",
                  "subPath": "SHIBBOLETH_KEY",
                },
              ],
            },
          ],
          "volumes": Array [
            Object {
              "name": "platform-shibboleth-cert",
              "secret": Object {
                "items": Array [
                  Object {
                    "key": "SHIBBOLETH_CERT",
                    "path": "SHIBBOLETH_CERT",
                  },
                ],
                "secretName": "platform",
              },
            },
            Object {
              "name": "platform-shibboleth-key",
              "secret": Object {
                "items": Array [
                  Object {
                    "key": "SHIBBOLETH_KEY",
                    "path": "SHIBBOLETH_KEY",
                  },
                ],
                "secretName": "platform",
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "RELEASE_NAME-platform",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "platform.pennlabs.org",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "RELEASE_NAME-platform",
                  "servicePort": 443,
                },
                "path": "/",
              },
            ],
          },
        },
      ],
      "tls": Array [
        Object {
          "hosts": Array [
            "platform.pennlabs.org",
          ],
          "secretName": "pennlabs-org-tls",
        },
      ],
    },
  },
  Object {
    "apiVersion": "cert-manager.io/v1alpha2",
    "kind": "Certificate",
    "metadata": Object {
      "labels": Object {
        "release": "RELEASE_NAME",
      },
      "name": "pennlabs-org",
    },
    "spec": Object {
      "dnsNames": Array [
        "pennlabs.org",
        "*.pennlabs.org",
      ],
      "issuerRef": Object {
        "group": "cert-manager.io",
        "kind": "ClusterIssuer",
        "name": "wildcard-letsencrypt-prod",
      },
      "secretName": "pennlabs-org-tls",
    },
  },
]
`;
